### PR TITLE
Smaller, more isolated agents, image family, no metadata-cache

### PR DESCRIPTION
**Pull request by fejta**
Reduce the size of our agents: 

* 8 cores per instance
* 10 light executors, single cpu-intensive executors
* Create agents based off an agent image
* Disable metadata cache on agents
* Share ssh keys between all agents

### CODE REVIEW

**Comment by fejta** [comment]
Next logical step would be to put things into an instance-template and then managed instance group, but this is the present state of the world.

File: jenkins/agent-ctl.sh [hunk 1]
```diff
@@ -92,28 +103,43 @@ fi
...
-    DISK_SIZE='200GB'
+    # Current experiment:
+    # 8 agents
+    # 1 executor, n1-highmem-8, 150G pd-standard
+    # Results:
+    # 14-32 cores, 12G ram, 150 write IOPs, <20MB/s write
+    DISK_SIZE='150GB'
     DISK_TYPE='pd-standard'
-    MACHINE_TYPE='n1-standard-16'
+    MACHINE_TYPE='n1-standard-8'
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: I think this may be the cause of kubernetes/kubernetes#27444.

Comment (spxtr) [inline jenkins/agent-ctl.sh]: If we only have a handful of heavy post-submit agents I think we can switch to n1-standard-32.

Comment (fejta) [inline jenkins/agent-ctl.sh]: Please try highcpu-16 first

**Pull request merged**